03/app.js

```diff
@@ -5,17 +5,65 @@ import Category from './Category';
 import Cart from './Cart';
 
 import data from './data.json';
+import Product from './Product';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
-    
+    addItem = (product) => {
+        this.setState(state => {
+            return {
+                cart: [...state.cart, product]
+            }
+        }) 
+    }
+    removeItem = (product) => {
+        this.setState(state => {
+            const newCart = state.cart.filter(element => element.id !== product.id);
+            return {cart: newCart}
+        })
+    }
     render() {
         return (
             <section>
-                <Category />
-                <Cart />
+                <Category>
+                    {data.map(product => {
+                            if(this.state.cart.find(e => e.id === product.id)){
+                                return (
+                                    <Product 
+                                    key={product.id}
+                                    product={product}
+                                    isCart={true} 
+                                    remove={this.removeItem}
+                                />
+                                )
+                            }else{
+                                return (
+                                    <Product 
+                                    key={product.id}
+                                    product={product}
+                                    isCart={false} 
+                                    isCategory={true}
```

Comment:
Czy nie można po prostu mieć jednej "zmiennej"? Jeśli nie jest "cart" to jest "category" :)